### PR TITLE
feat: expose function to get table of add actions

### LINKED ISSUE #1031: Expose the Delta Log in a DataFrame that's easy for analysis
The _delta_log contains all sorts of valuable information for end users.  Valuable chunks of Delta Log data are stored in JSON files that aren't easy for users to access.

It'd be great if the file name, file size, modification time, and column statistics were exposed to the user in a DataFrame so they could better manage their Delta table.  Here's a possible interface:

```python
import deltalake as dl

dt = dl.DeltaTable("./tmp/delta-table")
dt.delta_log_detail()
```

That would return a DataFrame with these columns:

* `file_name`
* `file_size`
* `modification_time`
* `data_change`
* `col_a_min`
* `col_a_max`
* `col_b_min`
* `col_b_max`
* ...

Here are the types of questions the users could answer with this metadata:

* How many files in my Delta table have less than 10,000 bytes?
* What’s the current distribution of the col_a_max values?
* How many bytes of data did we ingest yesterday?

This would help users a lot before they perform expensive computations.

### PR DESCRIPTION
**Pull request by wjones127**
# Description

Exposes function to get a dataframe of add actions for selected version of the table. 

TODO:

 * [x] add unit tests
 * [x] write user guide
 * [x] handle partition columns
 * [x] handle stats
 * [x] handle tags
 * [x] add a `flatten` option

# Related Issue(s)

- closes #1031

# Documentation

### CODE REVIEW

**Comment by chitralverma** [comment]
@wjones127 Can we have an indicator of "number of version available" to this metadata some where?

**Comment by wjones127** [comment]
Example:

```python
In [1]: from deltalake import DeltaTable, write_deltalake

In [2]: import pyarrow as pa

In [3]: data = pa.table({"x": [1, 2, 3], "y": [4, 5, 6]})

In [4]: write_deltalake("tmp", data, partition_by=["x"])

In [5]: dt = DeltaTable("tmp")

In [6]: dt.get_add_actions_df()
Out[6]: 
pyarrow.RecordBatch
path: string
size_bytes: int64
modification_time: timestamp[ms]
data_change: bool
partition_values: struct<x: int64>
  child 0, x: int64
num_records: int64
null_count: struct<y: int64 not null>
  child 0, y: int64 not null
min: struct<y: int64 not null>
  child 0, y: int64 not null
max: struct<y: int64 not null>
  child 0, y: int64 not null

In [7]: dt.get_add_actions_df().to_pandas()
Out[7]: 
                                                path  size_bytes       modification_time  data_change partition_values  num_records null_count       min       max
0  x=2/0-91820cbf-f698-45fb-886d-5d5f5669530b-0.p...         565 1970-01-20 08:40:08.071         True         {'x': 2}            1   {'y': 0}  {'y': 5}  {'y': 5}
1  x=3/0-91820cbf-f698-45fb-886d-5d5f5669530b-0.p...         565 1970-01-20 08:40:08.071         True         {'x': 3}            1   {'y': 0}  {'y': 6}  {'y': 6}
2  x=1/0-91820cbf-f698-45fb-886d-5d5f5669530b-0.p...         565 1970-01-20 08:40:08.071         True         {'x': 1}            1   {'y': 0}  {'y': 4}  {'y': 4}

In [8]: dt.get_add_actions_df(flatten=True).to_pandas()
Out[8]: 
                                                path  size_bytes       modification_time  data_change  partition.x  num_records  null_count.y  min.y  max.y
0  x=2/0-91820cbf-f698-45fb-886d-5d5f5669530b-0.p...         565 1970-01-20 08:40:08.071         True            2            1             0      5      5
1  x=3/0-91820cbf-f698-45fb-886d-5d5f5669530b-0.p...         565 1970-01-20 08:40:08.071         True            3            1             0      6      6
2  x=1/0-91820cbf-f698-45fb-886d-5d5f5669530b-0.p...         565 1970-01-20 08:40:08.071         True            1            1             0      4      4
```

File: rust/src/lib.rs [hunk 1]
```diff
@@ -70,6 +70,7 @@
 
 #![deny(warnings)]
 #![deny(missing_docs)]
+#![allow(rustdoc::invalid_html_tags)]
```
Comment (wjones127) [inline rust/src/lib.rs:73]: I need this in order to build the docs for some reason. Newer lint?

**Review by roeap — DISMISSED** [review]
Great work! Really looking forward to moving arrow deeper into our log handling :).

Left one minor namin comment that you may want to look at, otherwise LGTM!

File: python/deltalake/table.py [hunk 2]
```diff
@@ -440,3 +440,37 @@ def __stringify_partition_values(
                 str_value = str(value)
             out.append((field, op, str_value))
         return out
+
+    def get_add_actions_df(self, flatten: bool = False) -> pyarrow.RecordBatch:
```
[the remark this answers is not in the document]
Comment (wjones127) [reply python/deltalake/table.py:444]: I'm open to that. What do you think @MrPowers?

**Review by roeap — DISMISSED** [review]

File: python/deltalake/table.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MrPowers) [reply python/deltalake/table.py:444]: I think `get_add_actions` and `to_pandas()` is fine.  I'm not the best authority for the Pythonic way of doing things 😉

**Review by roeap — DISMISSED** [review]

**Review by roeap — APPROVED** [review]

**Pull request merged**